staging/src/k8s.io/apiserver/pkg/quota/v1/generic/evaluator.go

```diff
@@ -199,6 +199,11 @@ func CalculateUsageStats(options quota.UsageStatsOptions,
 		// need to verify that the item matches the set of scopes
 		matchesScopes := true
 		for _, scope := range options.Scopes {
+			// Specifying "PriorityClass" as scope is a no-op, as a Pod is always associated with a PriorityClass.
+			// The effective check will happen in the "ScopeSelector" later.
+			if scope == corev1.ResourceQuotaScopePriorityClass {
```

Comment:
rather than special-casing ResourceQuotaScopePriorityClass here, shouldn't we fix podMatchesScopeFunc instead to understand ResourceQuotaScopePriorityClass? that would make this block match the scope, and then the ScopeSelector block below would also run

Reply:
SG. Updated.